ffmpeg/extras.c

```diff
@@ -130,6 +130,29 @@ int lpms_rtmp2hls(char *listen, char *outf, char *ts_tmpl, char* seg_time, char
   return ret == AVERROR_EOF ? 0 : ret;
 }
 
+//Calculates the duration of audio stream by counting the number of audio packets
+double calculate_audio_dur(AVFormatContext *ic, int astream) {
+  AVPacket pkt;
+  av_init_packet(&pkt);
+  double duration = 0;
+  int64_t last_pts = AV_NOPTS_VALUE;
+  while (av_read_frame(ic, &pkt) >= 0) {
```

Comment:
this will require reworking if another part of GetCodecInfo ever requires another pass through the file

Reply:
I added `av_seek_frame` to reset the file position. Will this work? https://github.com/livepeer/lpms/pull/407/files#diff-998114575433dc60658ff6542df0970e08cef7970396134aeedf868b90aace10R140

Reply:
Removed `calculate_stream_duration`